pyproject.toml

```diff
@@ -144,6 +145,9 @@ combine-as-imports = true
 "tests/providers/elasticsearch/log/elasticmock/__init__.py" = ["E402"]
 "tests/providers/elasticsearch/log/elasticmock/utilities/__init__.py" = ["E402"]
 
+[tool.ruff.flake8-tidy-imports.banned-api]
+"airflow.AirflowException".msg = "Use airflow.exceptions.AirflowException instead."
```

Comment:
Should we remove this https://github.com/apache/airflow/blob/117e40490865f04aed38a18724fc88a8cf94aacc/airflow/__init__.py#L84 then?

Reply:
I've also thought about it and I remembered that I used this import myself from the first days with Airflow (1.10.4 will remain in my heart forever) as DAD Author.

Better what we could do it deprecate it because in general this import is useless, even for end-users the reason is simple all other exceptions such as `AirflowSkipException`, `AirflowFailException` do-not lazy loaded in `airflow.__init__`

Reply:
Cool, I'm happy with deprecation

Reply:
I could deprecate it in this PR or in the separate one. Which option is better?

Reply:
A separate one will have be better